docs/coreneuron/how-to/coreneuron.md

```diff
@@ -0,0 +1,83 @@
+# NEURON - CoreNEURON Integration
+
+**CoreNEURON** is a compute engine for the **NEURON** simulator optimised for both memory usage and computational speed. Its goal is to simulate large cell networks with minimal memory footprint and optimal performance.
+
+If you are a new user and would like to use **CoreNEURON**, this tutorial will be a good starting point to understand the complete workflow of using **CoreNEURON** with **NEURON**.
+
+
+# How to install CoreNEURON
+
+**CoreNEURON** is a submodule of the **NEURON** repository and can be installed with **NEURON** by enabling the flag **NRN_ENABLE_CORENEURON** in the CMake command used to install **NEURON**.
+For example:
+
+    cmake .. \
+         -DNRN_ENABLE_INTERVIEWS=OFF \
+         -DNRN_ENABLE_MPI=OFF \
+         -DNRN_ENABLE_RX3D=OFF \
+         -DNRN_ENABLE_CORENEURON=ON
+
+## Notes
+
+### Performance
+**CoreNEURON** is by itself an optimized compute engine of **NEURON**, however to unlock better performance from the used CPUs it's suggested using either an **Intel Compiler** or the **NMODL ISPC Backend** which is described below. Those compilers are able to vectorize better the code than **GCC** achieving the best performance gains possible.
+
+### GPU execution
+**CoreNEURON** supports also GPU execution based on an **OpenACC** backend. To be able to use it it's needed to install **NEURON** and **CoreNEURON** with a compiler that supports **OpenACC**. Currently the best supported compiler for the **OpenACC** backend is **PGI** and this is the recommended one for compilation.
+To enable the GPU backend specify the following *CMake* variables:
+
+    -DNRN_ENABLE_CORENEURON=ON
+    -DCORENRN_ENABLE_GPU=ON
+
+### NMODL
+The **NMODL** Framework is a code generation engine for the **N**EURON **MOD**eling **L**anguage. 
+**NMODL** is a part of **CoreNEURON** and can be used to generate optimized code for modern compute architectures including CPUs and GPUs.
+**NMODL** can also generate code using a backend for the **ISPC** compiler, which can vectorize greatly the generated code running on CPU and accelerate further the simulation.
+
+> Disclaimer: To use the ISPC compiler backend ISPC compiler must be preinstalled in your system. More information on how to install ISPC can be found [here](https://ispc.github.io/downloads.html). 
```

Comment:
@iomaganaris : If mention NMODL option, I think we should mention build dependencies required. (e.g. jinja2, pyyaml, sympy etc).

Reply:
can't those be auto-installed somehow?

Reply:
Instead of adding more detailed information on this page, maybe I can just point out the following page that has all the information to install `NMODL`?
https://github.com/BlueBrain/nmodl/blob/master/INSTALL.md